src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseParticipationService.java

```diff
@@ -153,6 +153,38 @@ public ProgrammingExerciseStudentParticipation findTeamParticipationByExerciseAn
         return participationOptional.get();
     }
 
+    /**
+     * Tries to retrieve a programming exercise participation for the given programming exercise, username
+     *
+     * @param exercise        the programming exercise for which to find a participation.
+     * @param username        of the user to which the participation belongs.
+     * @param withSubmissions true if the participation should be loaded with its submissions.
+     * @return the participation for the given programming exercise and user.
+     * @throws EntityNotFoundException if there is no participation for the given exercise and user.
+     */
+    @NotNull
+    public ProgrammingExerciseStudentParticipation findStudentParticipationByExerciseAndStudentLoginOrThrow(ProgrammingExercise exercise, String username,
```

Comment:
Why can't we always use the version with `findFirst`? If there is only one real exam, it should not make a difference.

Reply:
you're right, fixed